server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -811,12 +815,31 @@ private void registerQueryParsers(List<SearchPlugin> plugins) {
         registerFromPlugin(plugins, SearchPlugin::getQueries, this::registerQuery);
     }
 
+    private void registerIntervalsSourceProviders(List<SearchPlugin> plugins) {
+        registerIntervalsSourceProvider(new SearchPlugin.IntervalSpec<>(IntervalsSourceProvider.Match.NAME,
```

Comment:
The `match` query is registered the same way so I wonder how the clash is handled internally ? We should maybe handle the source provider differently (e.g.: not as a named writeable) ?

Reply:
NamedWriteable is namespaced by class, so I think we're OK here

Reply:
different [namespace](https://github.com/elastic/elasticsearch/pull/36135/files#diff-134bbc9191b667aff10076bf32f166adR837), got it, thanks.